e2e/puppeteer-custom-environment.ts

```diff
@@ -16,24 +16,29 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
 
   // Take a screenshot right after failure
   async handleTestEvent(event, state) {
```

Comment:
When issue first happened in CircleCI, there were no screenshot taken. Without screenshot, it was hard to determine what happened. After some digging, issue was because it occurred inside `beforeEach`.
```
beforeEach(async () => {
    await signIn(page);
  });
```
Adding `case 'hook_failure'` solved problem.